core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java

```diff
@@ -83,83 +84,200 @@ public static SortedSet<String> listSchemas(Session session, Metadata metadata,
 
     public static SortedSet<String> listSchemas(Session session, Metadata metadata, AccessControl accessControl, String catalogName, Optional<String> schemaName)
```

Comment:
I would extract private method like `doListSchemas` and `handleConnectorErrros(() -> doListSchemas(...)`